src/main/java/org/opensearch/security/user/InternalUserProvider.java

```diff
@@ -0,0 +1,20 @@
+package org.opensearch.security.user;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+
+/**
+ * This interface will be defined in core following these changes. For now this is used for testing in the Security Plugin.
+ *
+ */
+public interface InternalUserProvider {
```

Comment:
What do you think about making this a 'ServiceAccountManager'?

Then it would have only the use cases needed in OpenSearch core, during extensions registration for find or create with enable / disable?